cli/command_ls.go

```diff
@@ -54,15 +54,14 @@ func (c *commandList) run(ctx context.Context, rep repo.Repository) error {
 }
 
 func (c *commandList) listDirectory(ctx context.Context, d fs.Directory, prefix, indent string) error {
-	entries, err := d.Readdir(ctx)
-	if err != nil {
-		return errors.Wrap(err, "error reading directory")
-	}
-
-	for _, e := range entries {
-		if err := c.printDirectoryEntry(ctx, e, prefix, indent); err != nil {
-			return errors.Wrap(err, "unable to print directory entry")
+	if err := d.IterateEntries(ctx, func(innerCtx context.Context, e fs.Entry) error {
+		if err2 := c.printDirectoryEntry(innerCtx, e, prefix, indent); err2 != nil {
```

Comment:
nit: in such cases simple return c.printDirectoryEntry() is usually enough, you may need to silence linter, which is fine.